public/app/features/datasources/components/picker/utils.test.ts

```diff
@@ -77,6 +77,16 @@ describe('getDataSouceCompareFn', () => {
     ] as DataSourceInstanceSettings[]);
   });
 
+  it('sorts favorite datasources first', () => {
+    dataSources.sort(getDataSourceCompareFn(undefined, [], [], ['c', 'a']));
+    expect(dataSources).toEqual([
+      { uid: 'a', name: 'a', meta: { builtIn: true } },
+      { uid: 'c', name: 'c', meta: { builtIn: false } },
```

Comment:
Interesting. Presumably `c` was favorited first, so it should be sorted higher than `a`?